src/storage/buffer_manager/buffer_manager.cpp

```diff
@@ -72,13 +79,25 @@ void EvictionQueue::clear(std::atomic<EvictionCandidate>& candidate) {
     KU_UNREACHABLE;
 }
 
-BufferManager::BufferManager(uint64_t bufferPoolSize, uint64_t maxDBSize)
+BufferManager::BufferManager(const std::string& databasePath, const std::string& spillToDiskPath,
+    uint64_t bufferPoolSize, uint64_t maxDBSize, VirtualFileSystem* vfs)
     : bufferPoolSize{bufferPoolSize}, evictionQueue{bufferPoolSize / PAGE_SIZE},
-      usedMemory{evictionQueue.getCapacity() * sizeof(EvictionCandidate)} {
+      usedMemory{evictionQueue.getCapacity() * sizeof(EvictionCandidate)}, vfs{vfs} {
     verifySizeParams(bufferPoolSize, maxDBSize);
     vmRegions.resize(2);
     vmRegions[0] = std::make_unique<VMRegion>(REGULAR_PAGE, maxDBSize);
     vmRegions[1] = std::make_unique<VMRegion>(TEMP_PAGE, bufferPoolSize);
+
+    // TODO(bmwinger): It may be better to spill to disk in a different location for remote file
+    // systems, or even in general.
+    // Ideally we want to spill to disk in some temporary location such
+    // as /var/tmp (not /tmp since that may be backed by memory). However we also need to be able to
+    // support multiple databases spilling at once (can't be the same file), and handle different
+    // platforms.
+    if (!main::DBConfig::isDBPathInMemory(databasePath) &&
+        dynamic_cast<LocalFileSystem*>(vfs->findFileSystem(spillToDiskPath))) {
```

Comment:
Maybe. But when would we want to use this feature for another type of filesystem?
I'm also not sure how we would describe it. `isSpillable` sounds more like a question of whether it's possible, but we're not skipping this for remote filesystems just because it's not possible, but because it's impractical since the performance is poor.

On the other hand maybe it's not impractical if you're using fast network storage on your local network, so maybe there's an argument to allowing any filesystem when not using the default path (though I don't know why you wouldn't use something like NFS to treat it as a local filesystem in that case).

Reply:
hmm i didn't make myself clear here. To clarify a bit more, I mainly meant if that is a better way to check if the vfs is local or not compared to the check using `dynamic_cast`. maybe the function should be `isLocalFS()`.